File: c/leetcode/287. Find the Duplicate Number.c


int findDuplicate(int *nums, int numsSize)
{
  int *arr = (int *)malloc(sizeof(int) * numsSize + 1);

  for (int i = 0; i < numsSize; i++)
  {
    if (arr[nums[i]] == -1)
      return nums[i];
    else
      arr[nums[i]] = -1;
  }

  return 1;
}

File: javascript/leetcode/287. Find the Duplicate Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function(nums) {
  let set = new Set();

  for (let i of nums) {
    if (set.has(i)) return i;
    else set.add(i);
  }

  return 1;
};
